src/fides/api/ops/service/privacy_request/consent_email_batch_service.py

```diff
@@ -0,0 +1,260 @@
+        )
+        if not privacy_requests.first():
+            logger.info(
+                "Skipping batch consent email send with status: {}",
+                ConsentEmailExitState.no_applicable_privacy_requests.value,
+            )
+            return ConsentEmailExitState.no_applicable_privacy_requests
+
+        conn_configs: Query = get_consent_email_connection_configs(session)
+        if not conn_configs.first():
+            requeue_privacy_requests_after_consent_email_send(privacy_requests, session)
+            logger.info(
+                "Skipping batch consent email send with status: {}",
+                ConsentEmailExitState.no_applicable_connectors.value,
+            )
+            return ConsentEmailExitState.no_applicable_connectors
+
+        batched_user_data: List[BatchedUserConsentData] = stage_resource_per_connector(
+            conn_configs
+        )
+        add_batched_user_preferences_to_emails(privacy_requests, batched_user_data)
+
+        if not any(
+            pending_email.batched_user_consent_preferences
+            for pending_email in batched_user_data
+        ):
+            requeue_privacy_requests_after_consent_email_send(privacy_requests, session)
+            logger.info(
+                "Skipping batch consent email send with status: {}",
+                ConsentEmailExitState.missing_required_data.value,
+            )
+            return ConsentEmailExitState.missing_required_data
+
+        try:
+            send_prepared_emails(session, batched_user_data, privacy_requests)
+        except MessageDispatchException as exc:
+            logger.error(
+                "Consent email send for connector failed with exception: '{}'",
+                exc,
+            )
+            return ConsentEmailExitState.email_send_failed
+
+    requeue_privacy_requests_after_consent_email_send(privacy_requests, session)
+    return ConsentEmailExitState.complete
+
+
+def requeue_privacy_requests_after_consent_email_send(
+    privacy_requests: Query, db: Session
+) -> None:
+    """After batch consent email send, requeue privacy requests from the post webhooks step
+    to wrap up processing and transition to a "complete" state.
+
+    Also cache on the privacy request itself that it is paused at the post-webhooks state,
+    in case something happens in re-queueing.
+    """
+    logger.info("Batched consent email send complete.")
+    logger.info("Queuing privacy requests from 'post_webhooks' step.")
+    for privacy_request in privacy_requests:
+        privacy_request.cache_paused_collection_details(
+            step=CurrentStep.post_webhooks,
+            collection=None,
+            action_needed=None,
+        )
+        privacy_request.status = PrivacyRequestStatus.paused
+        privacy_request.save(db=db)
+
+        queue_privacy_request(
+            privacy_request_id=privacy_request.id,
+            from_step=CurrentStep.post_webhooks.value,
+        )
+
```

Comment:
As a reminder, we currently have several "steps" that happen in sequence in the privacy request runner.  A privacy request can be resumed from one of these steps if earlier steps shouldn't be repeated. 

After emails are sent for consent, what made sense to me was to put the privacy request in a paused state, and then requeue it to wrap up processing from the "post webhooks" step.  This step runs any post webhooks if they exist.  After this, any email notifications are sent about privacy request completion and then the privacy request is put in a completed state.